src/orchestrator/src/lib.rs

```diff
@@ -152,21 +152,29 @@ pub struct ServiceEvent {
     pub time: DateTime<Utc>,
 }
 
+/// Why the service is not ready, if known
+#[derive(Debug, Clone, Copy, Serialize, Eq, PartialEq)]
+pub enum NotReadyReason {
+    OOMKilled,
+}
+
 /// Describes the status of an orchestrated service.
 #[derive(Debug, Clone, Copy, Serialize, Eq, PartialEq)]
 pub enum ServiceStatus {
     /// Service is ready to accept requests.
     Ready,
     /// Service is not ready to accept requests.
-    NotReady,
+    /// The inner element is `None` if the reason
+    /// is unknown
+    NotReady(Option<NotReadyReason>),
```

Comment:
instead of `Option<NotReadyReason>` what do you think about adding an `Unknown` variant to `NotReadyReason`? It would remove one layer of nesting?

Reply:
The "extra" layer of nesting matches the actual value we emit to SQL -- `NULL` corresponds to `None`, whereas an inhabited SQL value corresponds to `Some`.

Reply:
Ahh that makes sense!